src/chunk.js

```diff
@@ -408,20 +408,43 @@ class Chunks {
    * @private
    */
   execute_(idleDeadline) {
-    this.scheduledImmediateInvocation_ = false;
     const t = this.nextTask_(/* opt_dequeue */ true);
     if (!t) {
+      this.scheduledImmediateInvocation_ = false;
+      this.durationOfLastExecution_ = 0;
       return false;
     }
+    let before;
     try {
-      const before = Date.now();
-      this.timeSinceLastExecution_ = before;
+      before = Date.now();
       t.runTask_(idleDeadline);
-      dev().fine(TAG, t.getName_(), 'Chunk duration', Date.now() - before);
     } finally {
-      resolved.then(() => {
-        this.schedule_();
-      });
+      // We want to capture the time of the entire task duration including
+      // scheduled immediate (from resolved promises) micro tasks.
+      // Lacking a better way to do this we just scheduled 10 nested
+      // micro tasks.
+      resolved
```

Comment:
This is extremely hacky. I cannot come up with a better solution, but it's worth mentioning.

Reply:
Down the line, using the proposed APIs from https://github.com/WICG/main-thread-scheduling will hopefully make this code unnecessary.